articles/introduction-to-processors/index.md

```diff
@@ -0,0 +1,89 @@
+b8    04 00 00 00   #moving "print out" syscall number to eax
+cd    80            #calling the linux kernel to execute our print to stdout
+b8    01 00 00 00   #moving "sys_exit" call number to eax
+cd    80            #executing it via linux sys_call
+```
+
+
+
+To understand how code is turned from human-readable to machine-readable then executed, we shall use the __x86 Instruction Set Architecture__. This refers to a series of microprocessors developed by Intel based on the 8085 Processor. A __microprocessor__ is a programmable electronic chip that has computing and decision-making capabilities. These microprocessors(x86) share similar instruction set architectural models and are hence grouped. For example: 80186 & 80286 microprocessors.
+
+>An instruction set architecture refers to the basic design of the processor’s components. It represents how they have been organized and implemented and how they interact with main memory and other components. Architecture is concerned with the general plan and functions of a processor.
+
+## Overview of the x86 Processor
+
+![alt text](x86TrainerKit.png)
+
+
+
+The figure above represents the 8086 trainer kit. A trainer kit is a device that functions as a simulator of the actual 8086 processor. It is used as a tool for learning, teaching, and development.
+
+The 8086 is a 16-bit microprocessor that has 20 address lines and 16 data lines. An address bus is an internal channel that transfers the RAM location of the data being processed or the instruction being fetched. The number of lines (wires) in the address bus determines the maximum amount of RAM that the CPU can directly access as each line carries one bit of the address. Therefore, the x86 Processor has 2^(20) = 1,049,576 bytes = Approximately 1MB of RAM. 
+
+The 8086 microprocessor is divided into two functional parts: __Bus Interface Unit__(BIU) and the __Execution Unit__(EU). The Bus Interface Unit provides the interface of 8086 to external memory and I/O devices while the Execution Unit performs the operations and calculations as instructed by the computer program.
+
+
+The 8086 utilizes _pipelining_ to improve performance. This is a technique that balances the fetching and execution of an instruction, i.e., while an instruction is being executed, another is fetched. Pipelining increases the overall instruction throughput, thus speeds up the microprocessor.
+
+The Execution Unit triggers the start of the Bus Interface Unit by sending instructions. After finding the location required, it is accessed, and data is retrieved using the data bus. The Execution Unit then removes instructions from the Instruction Stream Byte Queue for processing. The Bus Interface Unit simultaneously fetches more instructions from memory and puts them in the queue.
+
+In this article, we have outlined the basic functionality of a processor. In the next article, we will dive into the details using the x86 Processor’s Instruction Set Architecture.

```

Comment:
Some more details in this article would add more value to the readers. Give an introduction here to the instruction set. In the next article, go in depth.